File: mRemoteV1/CS/Connection/Connection.InterfaceControl.cs
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
using mRemoteNG.App;


namespace mRemoteNG.Connection
{
	public partial class InterfaceControl
    {
        #region Public Properties
        private Connection.Protocol.Base _Protocol;
        public Connection.Protocol.Base Protocol
		{
			get { return this._Protocol; }
			set { this._Protocol = value; }
		}
			
		private Info _Info;
        public Info Info
		{
			get { return this._Info; }
			set { this._Info = value; }
		}
        #endregion
			
        #region Methods
		public InterfaceControl(Control Parent, Connection.Protocol.Base Protocol, Connection.Info Info)
		{
			try
			{
				this._Protocol = Protocol;
				this._Info = Info;
                this.Parent = Parent;
                this.Location = new Point(0, 0);
                this.Size = this.Parent.Size;
                this.Anchor = (System.Windows.Forms.AnchorStyles)(AnchorStyles.Bottom | AnchorStyles.Left | AnchorStyles.Right | AnchorStyles.Top);
				InitializeComponent();
			}
			catch (Exception ex)
			{
				Runtime.MessageCollector.AddMessage(Messages.MessageClass.ErrorMsg, "Couldn\'t create new InterfaceControl" + Constants.vbNewLine + ex.Message);
			}
		}
        #endregion
	}
}

File: mRemoteV1/Connection/Connection.InterfaceControl.vb
Imports mRemote3G.App
Imports mRemote3G.Connection.Protocol
Imports mRemote3G.Messages

Namespace Connection
    Public Class InterfaceControl

#Region "Properties"

        Private _Protocol As Base

        Public Property Protocol As Base
            Get
                Return Me._Protocol
            End Get
            Set
                Me._Protocol = value
            End Set
        End Property

        Private _Info As Info

        Public Property Info As Info
            Get
                Return Me._Info
            End Get
            Set
                Me._Info = value
            End Set
        End Property

#End Region

#Region "Methods"

        Public Sub New(Parent As Control, Protocol As Base, Info As Info)
            Try
                Me._Protocol = Protocol
                Me._Info = Info
                Me.Parent = Parent
                Me.Location = New Point(0, 0)
                Me.Size = Me.Parent.Size
                Me.Anchor = AnchorStyles.Bottom Or AnchorStyles.Left Or AnchorStyles.Right Or AnchorStyles.Top
                InitializeComponent()
            Catch ex As Exception
                Runtime.MessageCollector.AddMessage(MessageClass.ErrorMsg,
                                                    "Couldn't create new InterfaceControl" & vbNewLine & ex.ToString())
            End Try
        End Sub

#End Region
    End Class
End Namespace


